spyder/widgets/variableexplorer/tests/test_dataframeeditor.py

```diff
@@ -174,6 +179,34 @@ def test_change_format_with_format_not_starting_with_percent(qtbot, monkeypatch)
     with qtbot.assertNotEmitted(editor.sig_option_changed):
         editor.change_format()
 
+def test_header_bom():
+    if PY2:
+        df = read_csv(os.path.join(FILES_PATH, 'issue_2514.csv'))
+    else:
+        df = read_csv(os.path.join(FILES_PATH, 'issue_2514.csv'),
+                      encoding='utf-8-sig')
```

Comment:
@ccordoba12 I think that maybe more than only remove the condition for PY2 will be needed since the `to_text_string` method does nothing if the object is an instance of `str` in PY3. So, should we add a way to handle the BOM character for PY3?